### DISCUSSION TITLE
FAQ

### DISCUSSION THREAD

**Discussion by y377** [facelessuser/RegReplace#109] [Q&A] [answered]
Hello, author, please ask: I need to execute the command only once (I use the shortcut key) and replace it twice to achieve the following purposes:
First perform a search and replace at the top of the document. After the replacement, there is an empty field `date:`
Then search for a certain **“time”** in the document, such as `2023-5-1`;
Copy the found `2023-5-1` to the `date:` at the top of the document
The correct format is `date: 2023-05-01`
I checked the official documentation, but still can't figure out how to implement it. 
**Please give me some advice!** **thanks**

> English is not my native language. Please forgive me if the translation software caused me to fail to understand the official website documents.

**Comment by facelessuser** [comment]
Keep in mind, not every replace action is well suited for a simple find and replace, some may require you to write an actual script.

I don't really have enough information from what you are showing me to even guess as to whether your case is well suited for RegReplace.

At the very least, you would need to provide a minimal example: text buffer to operate on, regex rules you've tried along with the configured command, etc.

**Reply by y377** [reply]
please forgive my negligence!

> Regardless of whether this problem can be solved or not, `RegReplace` is the best in my heart. I still sincerely thank you for open source such an excellent tool!

I am using the default configuration after installing **RegReplace**, as shown below;

![image](https://github.com/facelessuser/RegReplace/assets/58632405/56405561-a5ed-421d-b4be-075e896a4dd6)

This is my shortcut key settings：

![image](https://github.com/facelessuser/RegReplace/assets/58632405/6b812926-54b4-4aed-a6c2-5dfdb1b1ee45)

--------
**Original document without `RegReplace`**

![image](https://github.com/facelessuser/RegReplace/assets/58632405/1f8ab273-eab7-4f68-80dc-b1ca1abd2f8b)

---------

**error：**
```json
        "头":
        {
            "find": "\\A\\s+^([\u4e00-\u9fa5].*)${自(\\d{4})年(\\d{1,2})月(\\d{1,2})日起施行}",
            "replace": "---\ntitle: \\1\ndate: \\2-\\3-\\4\ncategory: 行政法规\ndescription: \\1\ntag: \n---\n\n# \\1",
            "greedy": true,
            // "multi_pass": true
            // "format_replace": true
        },
```
**If you use the first configuration, it cannot handle the document header**

![image](https://github.com/facelessuser/RegReplace/assets/58632405/acf027cc-435c-4d65-a651-8d9a7548dd6f)

-----

```json
        "头":
        {
            "find": "\\A\\s+^([\u4e00-\u9fa5].*)$",
            "replace": "---\ntitle: \\1\ndate: \ncategory: 行政法规\ndescription: \\1\ntag: \n---\n\n# \\1",
            "greedy": true,
            // "multi_pass": true
            // "format_replace": true
        },
```

**If you only process the document header, it can complete the replacement very well, and there is an empty `date` field, as shown below**

![image](https://github.com/facelessuser/RegReplace/assets/58632405/76785f96-c8ee-4885-bd12-8b36f9334b4a)

---------

Please let me explain it to you in detail. I'd like to use a configuration similar to the first one, which handles both the document title and copying the year, month and day characters from the bottom of the document to the head. That is, the year, month, and day characters at the bottom of the document themselves are not modified in any way. They are just matched to the year, month, and day characters and then copied and pasted into the `date:` field at the head of the document (the date is formatted as `YYYY-MM-DD` format and paste it into the head of the document)

**Processing process diagram：**
![image](https://github.com/facelessuser/RegReplace/assets/58632405/2c76ec11-b912-41ca-b35f-9248d386c3cc)

------------------

**What I'm looking forward to:**

![image](https://github.com/facelessuser/RegReplace/assets/58632405/c7deda69-6980-4a67-9965-30a931c52f4b)

**Reply by facelessuser** [reply, marked as answer]
The only way to pull off what, in the way that you want to, is to capture both the header and the date in one search so you have access to all the required capture groups in one find/replace. This is a simple example that is similar but not the same.

<img width="1084" alt="Screenshot 2023-10-16 at 8 38 54 PM" src="https://github.com/facelessuser/RegReplace/assets/1055125/9984cc0a-f024-485c-94f1-5cbb0cdeb913">

While it may be possible, this is not really the most efficient approach.

RegReplace does not have a way to find something and copy it for use in later patterns. Everything that is to be replaced must also contain all the context needed for the replace.

To be honest, this particular case probably would be best done with a script written specifically for the task.

**Reply by y377** [reply]
Thanks for your answer, I will add this use case if I think of a way
